TombEngine/Game/effects/effects.cpp

```diff
@@ -1283,36 +1283,85 @@ void TriggerFireFlame(int x, int y, int z, FlameType type)
 
 	spark->on = true;
 
+	int colorsR = std::clamp(int(color1.x * UCHAR_MAX), 0, UCHAR_MAX);
+	int colorsG = std::clamp(int(color1.y * UCHAR_MAX), 0, UCHAR_MAX);
+	int colorsB = std::clamp(int(color1.z * UCHAR_MAX), 0, UCHAR_MAX);
+
+	int colordR = std::clamp(int(color2.x * UCHAR_MAX), 0, UCHAR_MAX);
+	int colordG = std::clamp(int(color2.y * UCHAR_MAX), 0, UCHAR_MAX);
+	int colordB = std::clamp(int(color2.z * UCHAR_MAX), 0, UCHAR_MAX);
+
+
 	if (type == FlameType::Small)
 	{
-		spark->sR = spark->sG = (GetRandomControl() & 0x1F) + 48;
-		spark->sB = (GetRandomControl() & 0x3F) - 64;
+		if (color1 != Vector3(1.0f, 0.2f, 0.2f) && color2 != Vector3(1.0f, 0.8f, 0.1f))
```

Comment:
This is extremely weird hardcoded condition for specific color.

Reply:
this check must have a value to distinguish between colored flame and normal flame. For example, if Lara has a colored fire effect and hits a flame emitter, or another TriggerFireSparks function is called, the Item->Effect.EffectColor1/2 is not updated and she keeps burning with color. I've now set the values to 0 for normal fire.